Q1. What are the three measures of central tendency?

#### The three main measures of central tendency are:
#### 1) Mean: The mean, often referred to as the average, is calculated by summing all values in a dataset and dividing by the number of observations.
#### 2) Median: The median is the middle value of a dataset when it is ordered. If there is an even number of observations, the median is the average of the two middle values.
#### 3) Mode: The mode is the value that occurs most frequently in a dataset.

Q2. What is the difference between the mean, median, and mode? How are they used to measure the
central tendency of a dataset?

### Differences:
#### 1) Sensitivity to Extreme Values:
#### Mean: Sensitive to extreme values. A few very high or low values can significantly impact the mean.
#### Median: Less sensitive to extreme values. The median is not influenced by outliers.
#### Mode: Not affected by extreme values, as it is determined solely by the most frequent value.
#### 2) Calculation:
#### Mean: Calculated by summing all values and dividing by the number of observations.
#### Median: The middle value when data is ordered, or the average of the two middle values for an even dataset.
#### Mode: The value(s) that occur most frequently.
#### 3) Applicability:
#### Mean: Suitable for symmetric distributions without significant skewness.
#### Median: Suitable for skewed distributions or datasets with outliers.
#### Mode: Useful for identifying the most common value, especially in categorical data.
### Usage in Measuring Central Tendency:
#### Single Measure: The mean provides a balance point, the median gives the middle value, and the mode identifies the most frequent value.
#### Consideration of Data Distribution: The choice between mean, median, or mode often depends on the distribution of the data. In symmetric distributions, the mean may be preferred, while skewed or non-normally distributed data may be better represented by the median.
#### Robustness: The median and mode are more robust to extreme values and outliers, making them suitable for datasets with skewed distributions or data points that significantly differ from the majority.

Q3. Measure the three measures of central tendency for the given height data:                   
[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [2]:
import numpy as np 

In [3]:
### Mean  
height=[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]
np.mean(height)

177.01875

In [4]:
### Median 
np.median(height)

177.0

In [6]:
### Mode
from scipy import stats
stats.mode(height)

/tmp/ipykernel_1950/27516264.py:3: FutureWarning: Unlike other reduction functions (e.g. `skew`, `kurtosis`), the default behavior of `mode` typically preserves the axis it acts along. In SciPy 1.11.0, this behavior will change: the default value of `keepdims` will become False, the `axis` over which the statistic is taken will be eliminated, and the value None will no longer be accepted. Set `keepdims` to True or False to avoid this warning.
  stats.mode(height)


ModeResult(mode=array([177.]), count=array([3]))

Q4. Find the standard deviation for the given data:                             
[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [8]:
new_height=[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [9]:
np.std(new_height)

1.7885814036548633

Q5. How are measures of dispersion such as range, variance, and standard deviation used to describe
the spread of a dataset? Provide an example.

### Measures of Dispersion:
#### 1) Range: The range is the difference between the maximum and minimum values in a dataset.
#### Use: Range provides a simple measure of the spread of the data. It is sensitive to extreme values but may not accurately represent overall variability.
#### 2) Variance: Variance measures the average squared deviation of each data point from the mean.
#### Use: Variance provides a measure of the overall variability in a dataset. However, it is in squared units, making interpretation less intuitive.
#### 3) Standard Deviation: The standard deviation is the square root of the variance.
#### Use: Like variance, the standard deviation measures the spread of data points. It has the advantage of being in the same units as the original data, making it more interpretable than variance.
#### How They Describe the Spread of a Dataset:
#### Example: Consider two datasets of exam scores:
Dataset A: [70, 72, 75, 78, 80]                   
Dataset B: [60, 70, 75, 85, 90]
#### Range: The range of Dataset A (10) is smaller than the range of Dataset B (30). The range indicates that Dataset A has less variability or spread.
#### Variance: The variance for Dataset A is 13.6 and Dataset B is 114. A larger variance suggests greater variability. The variance of Dataset B is higher, it indicates more spread in scores.
#### Standard Deviation: The standard deviation complements the variance. The standard deviation is larger for Dataset B, it indicates greater variability in scores compared to Dataset A.

Q6. What is a Venn diagram?

#### A Venn diagram is a graphical representation of the relationships between sets. It uses circles or other shapes to depict the extent of overlap or common elements between different sets. Each set is represented by a closed curve, and the overlapping regions show the elements that are common to more than one set.
#### Key features of a Venn diagram:
#### Sets: Each closed curve in a Venn diagram represents a set.
#### Overlapping Regions: The overlapping areas between sets represent the elements that are common to those sets.
#### Non-overlapping Regions: The non-overlapping portions of each set represent elements that are unique to that particular set.

Q7. For the two given sets A = (2,3,4,5,6,7) & B = (0,2,6,8,10). Find:           
(i) A intersect B      
(ii) A ⋃ B

In [4]:
A= {2,3,4,5,6,7}

In [5]:
B= {0,2,6,8,10}

In [8]:
### i) Intersection
A_int_B=A & B 
print( A_int_B)

{2, 6}


In [9]:
### ii)  Union 
A_uni_B=A | B
print(A_uni_B)

{0, 2, 3, 4, 5, 6, 7, 8, 10}


Q8. What do you understand about skewness in data?

#### Skewness is a measure of the asymmetry or lack of symmetry in a dataset's distribution. In a symmetrical distribution, the left and right sides are mirror images of each other. Skewness quantifies the extent and direction of any deviation from symmetrical distribution.
#### There are three types of skewness:
#### 1) Positive Skewness (Right Skewness): The right tail is longer or fatter than the left. The majority of the data points are concentrated on the left side of the distribution, and the right side extends further. The mean is usually greater than the median. It is also called "right skewness" because the tail points towards the right.
#### 2) Negative Skewness (Left Skewness): The left tail is longer or fatter than the right. The majority of the data points are concentrated on the right side of the distribution, and the left side extends further. The mean is usually less than the median. It is also called "left skewness" because the tail points towards the left.
#### 3) Zero Skewness: The distribution is perfectly symmetrical. The mean and median are equal, and the tails on both sides are of equal length.

Q9. If a data is right skewed then what will be the position of median with respect to mean?

#### Position of the Mean: The mean is influenced by extreme values or outliers on the right side of the distribution. It is typically greater than the median.
#### Position of the Median: The median represents the middle value when the data is ordered from smallest to largest. In a right-skewed distribution, the median is typically less than the mean. The median is less affected by extreme values compared to the mean.
#### So, in a right-skewed distribution, the position of the median is usually to the left of the mean. 

Q10. Explain the difference between covariance and correlation. How are these measures used in
statistical analysis?

### Difference : 
#### Covariance:
#### Definition: Covariance is a measure of how much two variables change together. It indicates whether an increase in one variable is associated with an increase or decrease in another variable. The sign of the covariance indicates the direction of the relationship (positive or negative), while the magnitude of covariance is not standardized and depends on the scales of the variables.
#### Interpretation:
##### Positive covariance: Indicates that as one variable increases, the other tends to increase.
##### Negative covariance: Indicates that as one variable increases, the other tends to decrease.
##### Covariance close to zero: Indicates a weak or no linear relationship.
#### Correlation:
#### Definition: Correlation is a standardized measure of the strength and direction of a linear relationship between two variables. Unlike covariance, correlation is scaled between -1 and 1, where -1 indicates a perfect negative linear relationship, 1 indicates a perfect positive linear relationship, and 0 indicates no linear relationship.
#### Interpretation:
##### r=1 : Perfect positive linear relationship.
##### r=−1: Perfect negative linear relationship.
##### r=0: No linear relationship.
##### −1<r<0: Negative correlation (as one variable increases, the other tends to decrease).
##### 0<r<1: Positive correlation (as one variable increases, the other tends to increase).
### Usage in Statistical Analysis:
#### Covariance:
##### i) Covariance is used to understand the direction of the relationship between two variables.
##### ii) It is not standardized and depends on the units of the variables.
##### iii) Covariance is not easily interpretable in terms of the strength of the relationship.
#### Correlation:
##### i) Correlation is widely used because it is standardized and allows for easier interpretation of the strength and direction of the relationship.
##### ii) The correlation coefficient (r) ranges from -1 to 1, making it comparable across different datasets.
##### iii) Correlation is more robust to differences in the scales of the variables, making it a preferred measure for comparing relationships.

Q11. What is the formula for calculating the sample mean? Provide an example calculation for a
dataset.

#### The sample mean is calculated by summing up all the individual data points in a sample and dividing the sum by the number of observations in the sample. The formula for the sample mean is as follows: 
#### sample mean = (sum of all Xi)/n 

#### Consider the following dataset of exam scores for a sample of students: [75, 82, 68, 90, 78]

In [2]:
sample_mean=(75+82+68+90+78)/5
print(sample_mean)

78.6


#### So, the sample mean for this dataset is 78.6. It represents the average exam score for the sample of students.

Q12. For a normal distribution data what is the relationship between its measure of central tendency?

#### Mean:
##### In a normal distribution, the mean is equal to the median.
##### The mean is located at the center of the distribution, and the normal distribution is symmetric around the mean.
#### Median:
##### The median of a normal distribution is equal to the mean.
##### Since the normal distribution is symmetric, the median is at the center, and the data is equally distributed on both sides.
#### Mode:
##### The mode of a normal distribution is also equal to the mean and median.
##### In a perfectly symmetrical normal distribution, every point on the curve is a mode, and there is only one mode at the peak of the distribution.
#### Hence, for a normal distribution: Mean=Median=Mode

Q13. How is covariance different from correlation?

### Covariance:
#### Definition: Covariance measures how much two variables change together. It indicates the direction of the linear relationship between two variables (positive or negative) but does not provide a standardized measure of the strength of the relationship.
#### Units: Covariance is in the original units of the variables. The magnitude of covariance depends on the scales of the variables.
#### Range: The range of covariance is unbounded (can take any real value).
### Correlation:
#### Definition: Correlation is a standardized measure of the strength and direction of a linear relationship between two variables. It scales the covariance by the product of the standard deviations of the two variables.
#### Units: Correlation is unitless and ranges from -1 to 1. The magnitude of correlation is not affected by the scales of the variables.
#### Range: The range of correlation is between -1 and 1, where -1 indicates a perfect negative linear relationship, 1 indicates a perfect positive linear relationship, and 0 indicates no linear relationship.

Q14. How do outliers affect measures of central tendency and dispersion? Provide an example.

### Measures of central tendency : 
#### 1) Mean:
##### Outliers can significantly impact the mean because the mean is sensitive to extreme values.
##### An outlier that is much higher or lower than the other data points can pull the mean in its direction.
##### Example: Consider the dataset [10, 15, 20, 25, 30, 200]. The mean is greatly influenced by the outlier (200), resulting in a higher mean.
#### 2) Median:
##### The median is less affected by outliers because it is not influenced by the actual values but rather their order.
##### An outlier does not have as much impact on the median as it does on the mean.
##### Example: Using the same dataset [10, 15, 20, 25, 30, 200], the median remains relatively unchanged because it is not affected by the value of the outlier.
#### 3) Mode:
##### Outliers generally do not affect the mode, as the mode is determined by the most frequently occurring value.
##### If there is no repeated value, or all values occur with the same frequency, the dataset is considered to have no mode.
##### Example: In the dataset [10, 15, 20, 25, 30, 200], there is no mode.
### Measures of Dispersion:
#### 1) Range:
##### Outliers can significantly impact the range, as it is the difference between the maximum and minimum values.
##### A single outlier can lead to a much larger range.
##### Example: In the dataset [10, 15, 20, 25, 30, 200], the range is heavily influenced by the outlier (200).
#### 2) Variance and Standard Deviation:
##### Outliers can have a substantial impact on variance and standard deviation because they involve squared differences from the mean.
##### Squaring the difference accentuates the effect of outliers on these measures.
##### Example: Using the dataset [10, 15, 20, 25, 30, 200], the variance and standard deviation are greatly influenced by the outlier.
#### 3) Interquartile Range (IQR):
##### IQR is less sensitive to outliers because it involves the range of the middle 50% of the data.
##### It is less affected by extreme values in the tails of the distribution.
##### Example: In the dataset [10, 15, 20, 25, 30, 200], the IQR is less affected by the outlier.